File: C/binarySearch.c
#include <stdio.h>
#include <stdlib.h>

int binarySearch(int a[],int l,int r,int x){
    int mid=(l+r)/2;
    while(r>=l) {
        if (a[mid] == x) {
            return mid;
        }

        if (a[mid] > x) {
            return binarySearch(a, l, mid - 1, x);
        }

        if (a[mid] < x) {
            return binarySearch(a, mid + 1, r, x);
        }
    }
    return -1;
}

int main() {

    int size,i,x;

    printf("Enter the number of elements ");   //size of array input from user
    scanf("%d",&size);

    int *a=(int*)calloc(size,sizeof(int));  //memory allocated for array dynamically

    printf("\nEnter the elements ina sorted order ") ;       //input of elements by the user
    for(i=0;i<size;i++){
        scanf("%d",&a[i]);
    }

    printf("Enter the element to be searched ");
    scanf("%d",&x);

    int pos= binarySearch(a,0,size,x);
    printf("Element was found at %d position ",pos+1);

    return 0;
}

File: Python/Binary_Search.py
a=list(map(int,input().split()))
x=int(input())
l=0
r=len(a)-1
m=(l+r)//2
while(l<=r):
	m=(l+r)//2
	if(a[m]==x):
		print('Element found at location',m+1)
		break
	else:
		if(x>a[m]):
			l=m+1
		if(x<a[m]):
			r=m-1
else:
	print('Element not found')
